lib/fairy/fairy.c

```diff
@@ -258,13 +259,34 @@ void Fairy_InitFile(FairyFileInfo* fileInfo, FILE* file) {
 
                     /* Ignore the leading "." */
                     if (strcmp(&shstrtab[currentSection.sh_name + 1], "text") == 0) {
-                        fileInfo->progBitsSizes[FAIRY_SECTION_TEXT] += ALIGN(currentSection.sh_size, 0x10);
+                        if (gUseRealAlignment) {
+                            // Ensure the next file will start at its correct alignment
+                            fileInfo->progBitsSizes[FAIRY_SECTION_TEXT] = ALIGN(fileInfo->progBitsSizes[FAIRY_SECTION_TEXT], currentSection.sh_addralign);
+                            fileInfo->progBitsSizes[FAIRY_SECTION_TEXT] += ALIGN(currentSection.sh_size, currentSection.sh_addralign);
```

Comment:
Is this how it actually works? It aligns the end as well as the beginning?

Reply:
From what I understand, yes. I'm not completely sure about the end, but I'm sure about the beginning.

Reply:
Hmm. I'm concerned that this is essentially untestable since the linker won't behave anyway.

Reply:
Well, I did some manual tests (by modifying the generated linker script by hand) the other day, and it seemed to work fine.
It would require some kind of weird workaround on mkldscript's side for supporting this

Reply:
Yeah, that's basically the issue: we don't know if it's even useable anyway.